### ISSUE TITLE
twisted.web server don't output access log

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8272]
|<img alt="nakamuray's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| nakamuray reported|
|-|-|
|Trac ID|trac#8272|
|Type|release blocker: regression|
|Created|2016-04-05 14:34:55Z|
|Branch|https://github.com/twisted/twisted/tree/dropped-factory-8272|

From twisted-16.1.0, twisted.web server don't output access log anymore.

Start web server through `twistd -n web --path .` and access to the server by `curl http://localhost:8080/`.

On twisted-16.0.0, it output like:

```
2016-04-05 23:20:58+0900 [-] "127.0.0.1" - - [05/Apr/2016:14:20:57 +0000] "GET / HTTP/1.1" 200 1326 "-" "curl/7.48.0"
```

But on 16.1.0, it don't output access log.

<details><summary>Searchable metadata</summary>

```
trac-id__8272 8272
type__release_blocker__regression release blocker: regression
reporter__nakamuray nakamuray
priority__highest highest
milestone__None None
branch__branches_dropped_factory_8272 branches/dropped-factory-8272
branch_author__hawkowl hawkowl
status__closed closed
resolution__fixed fixed
component__web web
keywords__None None
time__1459866895355266 1459866895355266
changetime__1459917897943240 1459917897943240
version__None None
owner__hawkowl hawkowl

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_1"></a>|@hawkowl commented|
|-|-|

(In [47149]) Branching to dropped-factory-8272.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_2"></a>|@hawkowl set owner to @hawkowl|
|-|-|

Hi, thanks for the bug report.

This is a 16.1 regression related to some of the HTTP/2 work. The fix is minor, I'll have it up soon, and we'll roll a 16.1.1 when it is merged.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_3"></a>|@hawkowl set owner to @Lukasa|
|-|-|

Builders spun, please review.

**Comment by twisted-trac** [comment]
|[<img alt="Lukasa's avatar" src="https://avatars.githubusercontent.com/u/1382556?s=50" width="50" height="50">](https://github.com/Lukasa)<a name="note_4"></a>|@Lukasa set owner to @hawkowl|
|-|-|

Ok, this patch looks fine to me. It's a moderate shame that we miss a line of coverage here: it might be nice to write a test that validates that the getter works, but it's hard to know whether that just counts as absurdly trivial, so I'll leave that to your discretion at this time.

You've introduced new twistedchecker errors as well: please add an @ivar to the docstring for the class for _factory.

The new test is optional, the twistedchecker failure isn't. Back to you!

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_5"></a>|@hawkowl set status to `closed`|
|-|-|

(In [47173]) Merge dropped-factory-8272: Restore the factory being given to HTTPChannel

Author: hawkowl
Reviewer: lukasa
Fixes: [#8272](https://github.com/twisted/twisted/issues/8272)

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_6"></a>|@hawkowl commented|
|-|-|

(In [47175]) Merge dropped-factory-8272: Restore the factory being given to HTTPChannel

Author: hawkowl
Reviewer: lukasa
Fixes: [#8272](https://github.com/twisted/twisted/issues/8272)
